### DISCUSSION TITLE
Random polka dots on a rectangle

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#4715] [Digital Image Processing]
![polkadots](https://user-images.githubusercontent.com/12046302/149647380-89ae2e59-9b82-403e-a6b6-c8a63ba37f00.jpg)

I want to create random sized and random colored dots with opacity on a rectangle (as a design)

Any help is appreciated,

Thanks

**Comment by fmw42** [comment]
Your dots are really not random in position or size or color. They seem to be just random slight offsets from a set size and position.  How was this image created?

**Reply by fmw42** [reply]
If you are on a Unix-like system (Linux, Mac OSX or Windows 10 Unix), you could use my randomclipart script at my web site http://www.fmwconcepts.com/imagemagick with a circle clip art file. In the following I used a heart shaped clip art file.

![barbara_randomclipart_heart_D64_d16_A45_am45_p50_random](https://user-images.githubusercontent.com/13458218/149648301-09b140c9-d7ad-42d4-b32d-924d1d8dc3e0.jpg)

**Reply by srini8080** [reply]
can we control the opacity?

**Reply by fmw42** [reply]
Not currently in my script. But I think it could be added fairly easily.

**Reply by fmw42** [reply]
Do you mean random opacity or fixed opacity.

**Reply by srini8080** [reply]
fixed is fine.

**Reply by fmw42** [reply]
I will look into it tomorrow or Monday.

**Reply by srini8080** [reply]
Thank you

**Reply by fmw42** [reply]
Download the script now and try it. 

I added a new argument for a range of opacities for random coloring (only), that is for -c random. You may specify two values for the range or one value for a fixed opacity.  

Also If you provide a single color, it now can have an opacity value. All icons will be that transparent color. That is for -c some_color_value

**Reply by srini8080** [reply]
I did not have 'jot'.  Downloaded jot source 3.1.0 and installed. Says no option "-r"

from the script jot -r 1 0 1 >/dev/null

can we use "seq". Comes bundled with all linux versions

Thanks,

**Reply by fmw42** [reply]
> NAME
>      jot -- print sequential or random data
> 
> SYNOPSIS
>      jot [-cnr] [-b word] [-w word] [-s string] [-p precision]
>          [reps [begin [end [s]]]]
> 
> DESCRIPTION
>      The jot utility is used to print out increasing, decreasing, random, or
>      redundant data, usually numbers, one per line.
> 
>      The following options are available:
> 
>      -r      Generate random data instead of the default sequential data.
> 
> 

see https://man.cx/jot

seq does not have random values in my version.

But the script should work using the ImageMagick internal random() generator.

**Reply by srini8080** [reply]
I used freebsd version of jot.

It worked. Thanks much
